### PR TITLE
docs: Add gallery for ROUGE

### PR DESCRIPTION
**Pull request by baskrahmer**
## What does this PR do?

Add gallery for ROUGE and show how it can be used for benchmarking decoding parameters.

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2607.org.readthedocs.build/en/2607/

### CODE REVIEW

File: examples/text/rouge.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+"""ROUGE
+===============================
+
+The ROUGE (Recall-Oriented Understudy for Gisting Evaluation) metric used to evaluate the quality of generated text.
+
+A major difference with Perplexity comes from the fact that ROUGE evaluates actual text, whereas Perplexity evaluates logits.
+"""
```
Comment (Borda) [inline examples/text/rouge.py:7]: Can we have some story here?

File: examples/text/rouge.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+# %%
+# Here's a hypothetical Python example demonstrating the usage of perplexity to evaluate a generative language model:
```
Comment (baskrahmer) [inline examples/text/rouge.py:10]: Perplexity should be ROUGE here

**Review by Borda — APPROVED** [review]

File: examples/text/rouge.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+# %%
+# Calculate the ROUGE of the generated text
+
+rouge = ROUGEScore()
+rouge(preds=[sample_text], target=[target_text])
+
+# %%
+# By default, the ROUGE score is calculated using a whitespace tokenizer. You can also calculate the ROUGE for the tokens directly:
+token_rouge = ROUGEScore(tokenizer=lambda text: tokenizer.tokenize(text))
+rouge(preds=[sample_text], target=[target_text])
```
Comment (stancld) [inline examples/text/rouge.py:41]:
```suggestion
token_rouge(preds=[sample_text], target=[target_text])
```

**Review by stancld — APPROVED** [review]

**Comment by stancld** [comment]
@baskrahmer Thanks a lot for the addition! 👏

**Pull request merged**